atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/ConsensusForgettingStore.java

```diff
@@ -56,13 +55,14 @@ default void mark(Set<Cell> cells) {
      *   another update operation has failed in the past non-deterministically
      *   3. The semantics of the update operation depend on the underlying implementation.
      */
-    void atomicUpdate(Cell cell, byte[] value) throws KeyAlreadyExistsException;
+    AtomicUpdateResult atomicUpdate(Cell cell, byte[] value);
 
     /**
      * Performs atomic updates on multiple cells. This call may or may not guarantee atomicity across cells
      * depending on the underlying implementation.
+     * @return atomic update result for each individual cell
      */
-    void atomicUpdate(Map<Cell, byte[]> values) throws KeyAlreadyExistsException;
+    AtomicUpdateResult atomicUpdate(Map<Cell, byte[]> values);
 
```

Comment:
Thing is this api only makes sense for pue where we actually use the bathed api in kvs. This one makes no sense for MCAS as we are submitting individual tasks to a batched anyway. I think we should just throw for MCASConsensusForgettingStore.